### ISSUE TITLE
Valgrind: Uninitialized data access in cfeed_artificial_table_backend

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#4918]
This was caught intermittently in the table fuzzer test while shutting down the servers:
```
==17856== Thread 26:
==17856== Conditional jump or move depends on uninitialised value(s)
==17856==    at 0x1F5FD84: cfeed_artificial_table_backend_t::maybe_remove_machinery() (cfeed_backend.cc:134)
==17856==    by 0x1F5F31A: _ZZN32cfeed_artificial_table_backend_tC1EvENKUlvE_clEv (cfeed_backend.cc:41)
==17856==    by 0x1F600BC: _ZNSt17_Function_handlerIFvvEZN32cfeed_artificial_table_backend_tC1EvEUlvE_E9_M_invokeERKSt9_Any_data (functional:1778)
==17856==    by 0x170F761: std::function<void ()()>::operator()() const (functional:2161)
==17856==    by 0x1717FA2: call_ringer(std::function<void ()()>) (timing.cc:93)
==17856==    by 0x1718DCD: _ZNSt5_BindIFPFvSt8functionIFvvEEES2_EE6__callIvIEILi0EEEET_OSt5tupleIIDpT0_EESt12_Index_tupleIIXspT1_EEE (functional:1146)
==17856==    by 0x1718D5D: void std::_Bind<void (*()(std::function<void ()()>))(std::function<void ()()>)>::operator()<, void>() (functional:1205)
==17856==    by 0x1718D33: callable_action_instance_t<std::_Bind<void (*()(std::function<void ()()>))(std::function<void ()()>)> >::run_action() (callable_action.hpp:31)
==17856==    by 0x174BF77: callable_action_wrapper_t::run() (runtime_utils.cc:42)
==17856==    by 0x1750A70: coro_t::run() (coroutines.cc:204)
==17856==
```

From a quick look at the code, the one variable in that expression for which I can't verify that it gets initialized is `last_subscriber_time`.

**Comment by danielmewes** [comment]
@VeXocide could you take a look at this one please?

**Comment by VeXocide** [comment]
@danielmewes -- of course, `last_subscriber_time` is not initialised.

**Comment by VeXocide** [comment]
Fixed in `next` via commit f72024cd5c0f655ac01c92f5537b4d83bcf3f23a, over-the-shoulder by @danielmewes.

**Comment by VeXocide** [comment]
Fixed in `v2.1.x` as well via commit 7ea44701c9ef3d35502318a8ebb440134cd831b3.
